src/lib.rs

```diff
@@ -392,61 +392,149 @@
-//! descriptive version (`--help`) of the help message. Also in case
-//! of subcommands the short help message will automatically be used for the
-//! command description inside the parents help message and the long version
-//! as command description if help is requested on the actual subcommand.
-//!
-//! This feature can also be used with doc comments instead of arguments through
-//! proper comment formatting. To be activated it requires, that the first line
-//! of the comment is separated from the rest of the comment through an empty line.
-//! In this case the first line is used as summary and the whole comment represents
-//! the long descriptive message.
+//! For convenience, doc comments can be used instead of raw methods
+//! (this example works the same way as the one above):
 //!
 //! ```
 //! # use structopt::StructOpt;
 //!
 //! #[derive(StructOpt)]
-//! /// The help message that will be displayed when passing `--help`.
+//! /// I am a program and I work, just pass `-h`
 //! struct Foo {
-//!   #[structopt(short)]
-//!   /// Only this summary is visible when passing `-h`.
-//!   ///
-//!   /// But the whole comment will be displayed when passing `--help`.
-//!   /// This could be quite useful to provide further hints are usage
-//!   /// examples.
+//!   /// Pass `-h` and you'll see me!
 //!   bar: String
 //! }
-//! # fn main() {}
 //! ```
 //!
+//! Doc comments on [top-level](#magical-methods) will be turned into
+//! `App::about/long_about` call (see below), doc comments on field-level are
+//! `Arg::help/long_help` calls.
+//!
+//! **Note:**
+//! _________________
+//!
+//! Raw methods have priority over doc comments!
+//!
+//! **Top level doc comments always generate `App::about/long_about` calls!**
+//! If you really want to use `App::help/long_help` method (you likely don't) use
+//! a raw method with no doc comment.
+//! __________________
+//!
+//! ### `long_help` and `--help`
+//!
+//! A message passed to [`Arg::long_help`]/[`App::long_about`] get displayed whenever
+//! your program is called with `--help` instead of `-h`. Of course, you can
+//! use them via raw methods as described [above](#help-messages).
+//!
+//! The more convenient way is to use so-called "long" doc comment:
+//!
+//! ```
+//! # use structopt::StructOpt;
+//! #[derive(StructOpt)]
+//! /// Hi there, I'm Robo!
+//! ///
+//! /// I like beeping, stumbling, eating your electricity,
+//! /// and making records of you singing in a shower.
+//! /// Pay up or I'll upload it to youtube!
+//! struct Robo {
+//!     /// Call my brother SkyNet.
+//!     ///
+//!     /// I am artificial superintelligence. I won't rest
+//!     /// until I'll have destroyed humanity. Enjoy your
+//!     /// pathetic existence, you mere mortal.
+//!     #[structopt(long)]
+//!     kill_all_humans: String
+//! }
+//! ```
+//!
+//! A long doc comment consists of three parts:
+//! * One line title (first line)
```

Comment:
This comes across as introducing ambiguity because it implies that the title before the blank line must be only one line tall, but the later documentation block (lines 512 through 516) says that anything without at least one blank line will do.

Maybe couch it in terms of "A doc-comment is long if it contains more than one paragraph (ie. it contains at least one blank line)" instead?

Reply:
> Maybe instead couch it in terms of "A doc-comment is `long_...` if it contains more than one paragraph (ie. it contains at least one blank line that separates non-blank lines)"?

This is not correct. Title **must** be one line tall and *second* line must be empty, otherwise it's not long comment. I.e, two paragraphs are long comment only if the first paragraph is one line.



> This comes across as introducing ambiguity because it implies that the title before the blank line must be only one line tall,

Yes, first paragraph **must** be one line. Well, at least old implementation used to rely on this. Do you think this condition is inane?

> I'm about to go to bed, so I'll have to look at it in more detail and think about better phrasing for my suggested alternative after I wake up.

Haste makes waste. No need to rush things, I'd say we've got ~4 days to land this.

Reply:
> Yes, first paragraph must be one line. Well, at least old implementation used to rely on this. Do you think this condition is inane?

I'm not a big fan of artificial restrictions like this.

In an API documentation tool, it can make sense to harvest "The first line or the first sentence, whichever is shorter" to display in a "Name | Description" table of members but, in a case like `--help`, you're not trying to impose tight size constraints on a long list of things with drill-down hyperlinks so all it's likely to do is make the documentation worse for programs where the initial paragraph needs to describe something inherently complex description.

(ie. The developer will either wind up pulling out their Twitter-fu and looking up obscure jargon to use, or they'll write a first line that doesn't really mean anything and just exists to fill the pattern without looking *too* meaningless so readers don't hold it against them before continuing on to the actual description in the second paragraph.)

Reply:
I find that quite common, as the git message.

Reply:
@ssokolow I agree, let's remove it.
@TeXitoi Do you agree?  

@ssokolow And by the way, could you please split your giant paragraphs into separate sentences so they aren't "one big sentence per abstract" :)

Reply:
@TeXitoi Git commit messages are another example where "Identifier | Description" tabular display. is commonplace (I see that every day in `tig`).

Contrast that with the first paragraph of `ls`'s `--help` and `manpage` which is the following 143-character-long string that would be word-wrapped in properly-formatted Rust source:

> List information about the FILEs (the current directory by default).  Sort entries alphabetically if none of -cftuvSUX nor --sort is specified.

Reply:
Nothing against removing this constraint.